File: section14#Smoothing/C++/blur.cpp
#include <iostream>
#include <string>
#include <opencv2/opencv.hpp>


using namespace std;
using namespace cv;

#define KERNEL_THRESHOLD	200

void blurCallBack(int, void *);


int gKernelSize = 20;
Mat gImgOri, gImgOut;

int main()
{
	gImgOri = imread("Fig0235(c)(kidney_original).tif");
	namedWindow("imgOri");
	imshow("imgOri", gImgOri);

	blur(gImgOri, gImgOut, Size(gKernelSize, gKernelSize));
	namedWindow("imgOut");
	imshow("imgOut", gImgOut);

	cv::createTrackbar("Kernel Size", "imgOut", &gKernelSize, KERNEL_THRESHOLD, blurCallBack);

	waitKey(0);
	return true;
}


void blurCallBack(int ,void *)
{
	if (gKernelSize <= 0)
		return;
	blur(gImgOri, gImgOut, Size(gKernelSize, gKernelSize));
	imshow("imgOut", gImgOut);
	cout << gKernelSize << endl;
}

File: section14#Smoothing/Python/blur.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# author：lowkeyway time:8/1/2019

import sys
import cv2 as cv


def main_func(argv):
    imgOri = cv.imread("Fig0235(c)(kidney_original).tif")
    cv.namedWindow("imgOri")
    cv.imshow("imgOri", imgOri)


    imgOut = cv.blur(imgOri, (20, 20))
    cv.namedWindow("imgOut")
    cv.imshow("imgOut", imgOut)

    cv.waitKey(0)
    cv.destroyAllWindows()


if __name__ == '__main__':
    main_func(sys.argv)